src/filters.py

```diff
@@ -62,7 +69,22 @@ def remove_repeats(word: str) -> str:
 
 
 def remove_endings(word) -> str:
-    return word
+    analysis = morf.analyse(word)
+
+    word_core = ""
+    try:
+        word_core = analysis[0][2][1]
+        if word_core == "":
+            raise Exception
+    except:
+        print("enable to remove ending")
+
+    return extract_lemma(word_core)
+
+
+def extract_lemma(_word):
```

Comment:
This function is unnecessary in my opinion. You can implement these 2 lines inside `remove_endings` function.